src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodDriver.cs

```diff
@@ -0,0 +1,190 @@
+        {
+            // do nothing
+        }
+
+        protected internal override PinValue Read(int pinNumber)
+        {
+            PinValue result = default(PinValue);
+            try
+            {
+                Process myProc = new Process();
+                myProc.StartInfo.FileName = GpioGet;
+                myProc.StartInfo.Arguments = $" {GpioChip0} {pinNumber}";
+                myProc.StartInfo.RedirectStandardOutput = true;
+                myProc.Start();
+                if (HandlProcessExit(myProc, $"{GpioGet} {GpioChip0} {pinNumber}", _waitMilliSeconds))
+                {
+                    string valueContents = myProc.StandardOutput.ReadToEnd();
+                    result = ConvertSysFsValueToPinValue(valueContents);
+                }
+            }
+            catch (UnauthorizedAccessException e)
+            {
+                throw new UnauthorizedAccessException("Reading a pin value requires root permissions.", e);
+            }
+            return result;
+        }
+        private PinValue ConvertSysFsValueToPinValue(string value)
+        {
+            PinValue result;
+            value = value.Trim();
+
+            switch (value)
+            {
+                case "0":
+                    result = PinValue.Low;
+                    break;
+                case "1":
+                    result = PinValue.High;
+                    break;
+                default:
+                    throw new ArgumentException($"Invalid Gpio pin value {value}");
+            }
+
+            return result;
+        }
+
+        protected internal override void RemoveCallbackForPinValueChangedEvent(int pinNumber, PinChangeEventHandler callback)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void SetPinMode(int pinNumber, PinMode mode)
+        {
+            if (mode == PinMode.InputPullDown || mode == PinMode.InputPullUp)
+            {
+                throw new PlatformNotSupportedException("This driver is generic so it does not support Input Pull Down or Input Pull Up modes.");
+            }
+            _pinNumperToPinModes[pinNumber] = mode;
+        }
+
+        protected internal override WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventType, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();  // TODO
+        }
+
+        protected internal override void Write(int pinNumber, PinValue value)
+        {
+            string stringValue = ConvertPinValueToSysFs(value);
+            string command = $"--mode=signal {GpioChip0} {pinNumber}={stringValue}";
+            try
+            {
+                Process myProc;
+                if (PinValue.Low == value)
+                {
+                    if (_pinNumperToProcess.ContainsKey(pinNumber))
+                    {
+                        myProc = _pinNumperToProcess[pinNumber];
+                        myProc.Kill();
```

Comment:
Good catch, yes WaitForExit is done in HandleProcessExit method, forgot to remove the call made here. But Kill() shouldn't got to HandleProcessExit method as that method called for 2 different purpose. if write value is high it will set the output high and should wait until setpin called with low, only in that case we should kill that process keeping the pin high